### DISCUSSION TITLE
Add own transform and save result with new key in Dataset

### DISCUSSION THREAD

**Discussion by BailiangJ** [Project-MONAI/MONAI#2700] [Q&A] [answered]
Hi,
is it possible to implement my own transform/function, and save the result in the Dataset set with a new key?

For example, I have a function that could segment the image, 

seg_func(image): return seg

And I want something like this
data["seg"] = seg_func(data["image"]), where "seg" is a new key,
in a e.g. ComposeTransform,

how could I implement this?

Thanks in advance!

**Comment by Nic-Ma** [comment, marked as answer]
Hi @BailiangJ ,

Thanks for your interest and experiments with MONAI.
I think you can refer to the `SplitChanneld` transform, it saves every channel into a new key:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/dictionary.py#L340

Thanks.

**Reply by BailiangJ** [reply]
Hi @Nic-Ma ,

Great, then I could write a similar class to this and implement what I want.

Thanks for your reply!
Best.
